content/articles/2019-06-rlang-0-4-0.Rmarkdown

```diff
@@ -0,0 +1,131 @@
+---
+title: 'rlang 0.4.0'
+author: Lionel Henry
+date: '2019-06-27'
+slug: rlang-0-4-0
+description: >
+  rlang 0.4.0 is now on CRAN!
+categories:
+  - package
+photo:
+  url: https://unsplash.com/photos/ar7ZWQ-r87g
+  author: Caitlin Wynne
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE, comment = "#>"
+)
+
+library(dplyr)
+```
+
+It is with great excitement that we announce the release of [rlang 0.4.0](https://rlang.r-lib.org) on CRAN. rlang is a toolkit for working with core R and Tidyverse features, and hosts the tidy evaluation framework. The full set of changes can be found in the [changelog](https://rlang.r-lib.org/news/index.html). In this article, we introduce the most important of these, the new tidy evaluation operator `{{`.
+
+
+## The good and bad of tidy evaluation
+
+Tidy eval powers packages like dplyr or tidyr. It makes it possible to manipulate data frame columns as if they were defined in the workspace:
+
+```{r, error = TRUE}
+gender
+mass
+
+starwars %>%
+  group_by(gender) %>%
+  summarise(mass_maximum = max(mass, na.rm = TRUE))
+```
+
+We call this syntax __data masking__. It is made possible by the R language and makes it easier to write and read code in interactive scripts. Unfortunately, it has the flip side of making it more difficult to reuse common patterns inside functions:
+
+```{r, error = TRUE}
+max_by <- function(data, var, by) {
+  data %>%
+    group_by(by) %>%
+    summarise(maximum = max(var, na.rm = TRUE))
+}
+
+starwars %>% max_by(mass, by = gender)
+```
+
+Technically, this is because data masked code needs to be _delayed_ and _transported_ to the data context. Behind the scene, dplyr verbs achieve this by capturing the blueprint of your code, and resuming its evaluation inside of the data mask. The example above fails because `group_by()` is capturing the wrong piece of blueprint. To solve this, tidy evaluation provides `enquo()` to delay a blueprint, and the surgery operator `!!` for modifying blueprints. The combination of using `enquo()` and `!!` is called the __quote-and-unquote__ pattern:
```

Comment:
```suggestion
Technically, this is because data-masked code needs to be _delayed_ and _transported_ to the data context. Behind the scenes, dplyr verbs achieve this by capturing the blueprint of your code, and resuming its evaluation inside of the data mask. The example above fails because `group_by()` is capturing the wrong piece of blueprint. To solve this, tidy evaluation provides `enquo()` to delay a blueprint, and the surgery operator `!!` for modifying blueprints. The combination of using `enquo()` and `!!` is called the __quote-and-unquote__ pattern:
```

Reply:
^^ just added a hyphen between data and masked for "data-masked code" (compound modifier 🤓 — I don't know why some stuff doesn't show up on GH's diff)

Reply:
> To solve this, tidy evaluation provides `enquo()` to delay a blueprint

Does this delay _making_ a blueprint or _evaluating_ it?

Reply:
I think it's most technically correct to say it delays the evaluation of the blueprint.